src/IndexRepository.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Algolia\Settings;
+
+use AlgoliaSearch\Json;
+use Illuminate\Support\Facades\File;
+use Laravel\Scout\Searchable;
+
+final class IndexRepository
+{
+    public function __construct()
+    {
+        // Ensure settings directory exists
+        if (! File::exists($path = $this->getBasePath())) {
+            File::makeDirectory($path);
+        }
+    }
+
+    public function getSettings($indexName)
+    {
+        return Json::decode(
+            File::get($this->getFilePath($indexName)),
+            true
+        );
+    }
+
+    /**
+     * @param string $fqn
+     *
+     * @return bool
+     */
+    public function validateClassSearchable($fqn)
```

Comment:
In the end, if we go for a class that read and write settings/synonyms/rules into files, I'm not sure this method really belongs here.